sources/api/sundog/src/main.rs

```diff
@@ -334,9 +368,10 @@ where
         // TODO: We can optimize by using a prefix trie here. But there are no satisfactory trie
         //  implementations I can find on crates.io. We can roll our own at some point if this
         //  becomes a bottleneck
-        if populated_settings
-            .iter()
-            .any(|k| k.starts_with_segments(setting.segments()))
+        if generator_object.skip_if_populated
+            && populated_settings
+                .iter()
+                .any(|k| k.starts_with_segments(setting.segments()))
         {
             debug!("Setting '{}' is already populated, skipping", setting);
```

Comment:
It might be a good idea here to also emit a log message if the value is populated but `skip_if_populated` is true. One succinct way you could do that would be to match against a tuple of the values:

```
match (generator_object.skip_if_populated, populated_settings) {
    (true, true) => {
        debug!("Setting '{}' is already populated, skipping", setting);
        continue;
    },
    (false, true) => {
        debug!("Setting '{}' is already populated, but configured to be overwritten", setting);
    }
    _ => (),
}